metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -1054,49 +1054,66 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
             yield from self.gen_database_containers(database)
 
         for table in tables:
-            database_name = table["DatabaseName"]
             table_name = table["Name"]
-            full_table_name = f"{database_name}.{table_name}"
-            self.report.report_table_scanned()
-            if not self.source_config.database_pattern.allowed(
-                database_name
-            ) or not self.source_config.table_pattern.allowed(full_table_name):
-                self.report.report_table_dropped(full_table_name)
-                continue
+            try:
+                yield from self.gen_table_wu(table=table)
+            except KeyError as e:
+                self.report.report_failure(
+                    message="Failed to extract wu for table",
+                    context=f"Table: {table_name}",
+                    exc=e,
+                )
+        if self.extract_transforms:
+            yield from self._transform_extraction()
 
-            dataset_urn = make_dataset_urn_with_platform_instance(
-                platform=self.platform,
-                name=full_table_name,
-                env=self.env,
-                platform_instance=self.source_config.platform_instance,
-            )
+    def gen_table_wu(self, table: Dict) -> Iterable[MetadataWorkUnit]:
```

Comment:
Although we may not be very consistent with this...

```suggestion
    def _gen_table_wu(self, table: Dict) -> Iterable[MetadataWorkUnit]:
```

It seems we're currently wrapping the entire function in a `try/except` block, but the `KeyError` can only actually occur in the two lines where we access the `table` dictionary, correct? If that's the case, it would make sense to use a more targeted `try/except` block around those specific lines for better precision.

Reply:
done